src/plugins/intel_npu/src/compiler_adapter/src/plugin_compiler_adapter.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plugin_compiler_adapter.hpp"
+
+#include <ze_graph_ext.h>
+
+#include <memory>
+#include <string>
+
+#include "intel_npu/common/device_helpers.hpp"
+#include "intel_npu/common/itt.hpp"
+#include "intel_npu/config/compiler.hpp"
+#include "intel_npu/npu_private_properties.hpp"
+#include "intel_npu/utils/logger/logger.hpp"
+#include "intel_npu/utils/zero/zero_api.hpp"
+#include "intel_npu/utils/zero/zero_result.hpp"
+#include "openvino/util/file_util.hpp"
+#include "openvino/util/shared_object.hpp"
+#include "plugin_graph.hpp"
+#include "ze_graph_ext_wrapper.hpp"
+
+namespace {
+std::shared_ptr<void> loadLibrary(const std::string& libpath) {
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    return ov::util::load_shared_object(ov::util::string_to_wstring(libpath).c_str());
+#else
+    return ov::util::load_shared_object(libpath.c_str());
+#endif
+}
+
+std::shared_ptr<intel_npu::ICompiler> getCompiler(std::shared_ptr<void> so) {
+    static constexpr auto CreateFuncName = "CreateNPUCompiler";
+    auto symbol = ov::util::get_symbol(so, CreateFuncName);
+
+    using CreateFuncT = void (*)(std::shared_ptr<intel_npu::ICompiler>&);
+    const auto createFunc = reinterpret_cast<CreateFuncT>(symbol);
+
+    std::shared_ptr<intel_npu::ICompiler> compilerPtr;
+    createFunc(compilerPtr);
+    return compilerPtr;
+}
+
+ov::SoPtr<intel_npu::ICompiler> loadCompiler(const std::string& libpath) {
+    auto compilerSO = loadLibrary(libpath);
+    auto compiler = getCompiler(compilerSO);
+
+    return ov::SoPtr<intel_npu::ICompiler>(compiler, compilerSO);
+}
+}  // namespace
+
+namespace intel_npu {
+
+PluginCompilerAdapter::PluginCompilerAdapter(const std::shared_ptr<ZeroInitStructsHolder>& zeroInitStruct)
```

Comment:
Have you considered other names? Like `MLIRCompilerAdapter` to match the config value name.